python_modules/dagster/dagster_tests/asset_defs_tests/test_decorators.py

```diff
@@ -1375,3 +1376,15 @@ def _(): ...
 
     except DagsterInvalidDefinitionError as err:
         assert "except for Ins that have the Nothing dagster_type" not in str(err)
+
+
+def test_multi_asset_multiple_compute_kind():
+    with pytest.raises(
+        DagsterInvalidDefinitionError, match="Cannot specify multiple compute kinds"
+    ):
+
+        @multi_asset(
+            specs=[AssetSpec("foo", compute_kind="alpha"), AssetSpec("bar", compute_kind="beta")]
+        )
+        def foo_bar():
+            pass
```

Comment:
add test case where one of the specs has a kind and the other one is None

Reply:
And then the question what the policy should be. I think we should *require* it set on all of them since that is what is going to be appear in the UI.